### ISSUE TITLE
repository cloning

### ISSUE THREAD

**Issue by freddyomics** [guiwitz/napari-serialcellpose#4]
Hi Guillaume,

I couldn't install the package directly using:
```
pip install git+https://github.com/guiwitz/napari-serialcellpose.git
``` 

I got this error: 
```
pip install https://github.com/guiwitz/napari-serialcellpose.git
Collecting https://github.com/guiwitz/napari-serialcellpose.git
  Downloading https://github.com/guiwitz/napari-serialcellpose.git
     - 200.3 kB 4.0 MB/s 0:00:00
  ERROR: Cannot unpack file C:\Users\KBH Universitet\AppData\Local\Temp\pip-unpack-rree23eb\napari-serialcellpose.git (downloaded from C:\Users\KBH Universitet\AppData\Local\Temp\pip-req-build-j4ym_r08, content-type: text/html; charset=utf-8); cannot detect archive format
ERROR: Cannot determine archive format of C:\Users\KBH Universitet\AppData\Local\Temp\pip-req-build-j4ym_r08
```

After installing git separately and installing the github package, I got this error:
```
Collecting https://github.com/guiwitz/napari-serialcellpose.git
  Downloading https://github.com/guiwitz/napari-serialcellpose.git
     - 200.3 kB 4.0 MB/s 0:00:00
  ERROR: Cannot unpack file C:\Users\KBH Universitet\AppData\Local\Temp\pip-unpack-rree23eb\napari-serialcellpose.git (downloaded from C:\Users\KBH Universitet\AppData\Local\Temp\pip-req-build-j4ym_r08, content-type: text/html; charset=utf-8); cannot detect archive format
ERROR: Cannot determine archive format of C:\Users\KBH Universitet\AppData\Local\Temp\pip-req-build-j4ym_r08
```

I fixed it by using this code instead:
```
conda install -c conda-forge git pip
pip install git+https://github.com/guiwitz/napari-serialcellpose#egg=napari-serialcellpose`
```
Maybe you know why I had to adjust the installation.

**Comment by guiwitz** [comment]
> I got this error:
pip install https://github.com/guiwitz/napari-serialcellpose.git
Collecting https://github.com/guiwitz/napari-serialcellpose.git
  Downloading https://github.com/guiwitz/napari-serialcellpose.git
     - 200.3 kB 4.0 MB/s 0:00:00
  ERROR: Cannot unpack file C:\Users\KBH Universitet\AppData\Local\Temp\pip-unpack-rree23eb\napari-serialcellpose.git (downloaded from C:\Users\KBH Universitet\AppData\Local\Temp\pip-req-build-j4ym_r08, content-type: text/html; charset=utf-8); cannot detect archive format
ERROR: Cannot determine archive format of C:\Users\KBH Universitet\AppData\Local\Temp\pip-req-build-j4ym_r08

From your error message above, it looks like you used ```pip install https://github.com/guiwitz/napari-serialcellpose.git``` instead of ```pip install git+https://github.com/guiwitz/napari-serialcellpose.git ```. Can that be or is the error message wrong?

Also beware I’ll very soon update the plugin with some heavy changes making it more flexible. So keep an eye on updates.

**Comment by freddyomics** [comment]
I ran `pip install git+https://github.com/guiwitz/napari-serialcellpose.git`, but I guess the pip installation didn't work. I tried your  instructions now on a different computer and everything worked fine, so it seems like it was related to the computer. 

Good to hear that you are planning to update the package! I noticed that I couldn't select the individual channels of my TIF images but only the whole image. I also couldn't set a negative flow threshold. Are there updates regarding that?

**Comment by guiwitz** [comment]
Hi @freddyomics,

I just officially released the plugin officially on PyPi, so in principle now you should be able to install directly with ```pip install napari-serialcellpose```. Just have a look at the Readme as many things changes in this version 0.2.0.

In particular, if your images open as separate layers for each channel (via aicsimageio) you should be able to select them now. Regarding the flow threshold, I always assumed that the values vas between 0-1. What would a negative value mean?

**Comment by freddyomics** [comment]
Thanks! I installed the new version. 

I got an error now when selecting an image from the data folder:

`ValueError: There is no registered plugin named 'napari-aicsimageio'.
Names of plugins offering readers are: {'builtins', 'bfio', 'aicsimageio-out-of-memory', 'ome-types', 'aicsimageio-in-memory'}`

I can manually import the same image with aicsimageio-in-memory, but it still does not let me select the channels but only "None" and "Image:0".

I could set a negative flow threshold value in the cellpose plugin and it yielded the best result. The range is also bigger than 0-1.

**Comment by guiwitz** [comment]
The reason for that error is that the latest version of the napari-aicsimageio plugin has changed the name of the reader (to be compatible with the new plugin system) . So I guess you still have have a version of napari-aicsimageio<7.0.0 ? Did you install everything in a new environment? If not please try, as it should then install the latest version of that package. Let me know if it works. If not I'll try to fix this by allowing the use of older versions of napari-aicsimageio.

**Comment by freddyomics** [comment]
That was the problem! Thanks! I had an old version of napari-aicsimageio even though I created a new environment. I only managed to update it by downloading the latest version from github and not from PyPi or conda.

I can see the image now but the channels are still not accessible. I am not sure how it is coded in Cellpose, but there I can select 0=red, 1=green, and 2=blue.

**Comment by guiwitz** [comment]
What sort of image do you have? If your channels don't appear, it means it's not recognized as a multi-channel image by aicsimageio.

**Comment by freddyomics** [comment]
It's a TIF and the dims.shape is (1,1,1,2049,2049,3).

**Comment by guiwitz** [comment]
The default axis order in aicsimageio is "TCZYX" and if it interprets and image as being RGB (natural image) there's an additional dimension "TCZYXS". So it looks like aicsimageio opens your image as RGB which is why you can't select channels (during processing the image would just be converted to grayscale).

I made an attempt at giving users the choice of RGB or channels. In the options tab you have a check box "Do not treat as RGB" that you can try to use. It worked with a small example but no guarantees.

You can install that new experimental version with: 
```
pip install git+https://github.com/guiwitz/napari-serialcellpose.git@rgb_choice
```

Let me know if there's any improvement...

**Comment by freddyomics** [comment]
Yes, that worked! Now I can select the different channels and see them in the viewer. Unfortunately the segmentation still only results a blank mask and a ValueError related no NaNs:

```
ValueError                                Traceback (most recent call last)
File ~\Anaconda3\envs\serialcellpose\lib\site-packages\napari_serialcellpose\serial_widget.py:510, in SerialWidget._on_update_filtering_sliders(self=<napari_serialcellpose.serial_widget.SerialWidget object>)
    507 min_val = self.allprops[filtering_props].min()
    508 max_val = self.allprops[filtering_props].max()
--> 510 self.filterprop_min_slider.min = min_val
        min_val = nan
        self = <napari_serialcellpose.serial_widget.SerialWidget object at 0x000001F61D2215E0>
        self.filterprop_min_slider = FloatSlider(value=1.0, annotation=None, name='')
    511 self.filterprop_min_slider.max = max_val
    512 self.filterprop_max_slider.min = min_val

File ~\AppData\Roaming\Python\Python39\site-packages\magicgui\widgets\_bases\ranged_widget.py:90, in RangedWidget.min(self=FloatSlider(value=1.0, annotation=None, name=''), value=nan)
     88 @min.setter
     89 def min(self, value: float):
---> 90     self._widget._mgui_set_min(value)
        value = nan
        self._widget = <magicgui.backends._qtpy.widgets.FloatSlider object at 0x000001F633345E50>
        self = FloatSlider(value=1.0, annotation=None, name='')

File ~\AppData\Roaming\Python\Python39\site-packages\magicgui\backends\_qtpy\widgets.py:623, in Slider._mgui_set_min(self=<magicgui.backends._qtpy.widgets.FloatSlider object>, value=nan)
    621 def _mgui_set_min(self, value: float):
    622     """Set the minimum possible value."""
--> 623     super()._mgui_set_min(value)
        value = nan
    624     self._readout_widget.setMinimum(value)

File ~\AppData\Roaming\Python\Python39\site-packages\magicgui\backends\_qtpy\widgets.py:320, in QBaseRangedWidget._mgui_set_min(self=<magicgui.backends._qtpy.widgets.FloatSlider object>, value=nan)
    318 """Set the minimum possible value."""
    319 self._update_precision(minimum=value)
--> 320 val = self._pre_set_hook(value)
        value = nan
        self = <magicgui.backends._qtpy.widgets.FloatSlider object at 0x000001F633345E50>
    321 self._qwidget.setMinimum(val)

File ~\AppData\Roaming\Python\Python39\site-packages\magicgui\backends\_qtpy\widgets.py:686, in FloatSlider._pre_set_hook(self=<magicgui.backends._qtpy.widgets.FloatSlider object>, value=nan)
    685 def _pre_set_hook(self, value):
--> 686     return int(value * self._precision)
        value * self._precision = nan
        value = nan
        self._precision = 1000000.0
        self = <magicgui.backends._qtpy.widgets.FloatSlider object at 0x000001F633345E50>

ValueError: cannot convert float NaN to integer
C:\Users\AppData\Roaming\Python\Python39\site-packages\skimage\measure\_regionprops.py:594: RuntimeWarning: invalid value encountered in double_scalars!
C:\Users\AppData\Roaming\Python\Python39\site-packages\skimage\measure\_regionprops.py:594: RuntimeWarning: invalid value encountered in double_scalars!
C:\Users\Anaconda3\envs\serialcellpose\lib\site-packages\napari_serialcellpose\serial_analysis.py:169: UserWarning: Computing intensity features but no intensity image provided. Result will be zero.!
```

**Comment by guiwitz** [comment]
Did you select the channel to be segmented? Which model are you using for segmentation? Any chance you could share an example dataset privately? The error above is just a consequence of the fact that no objects are found (I will try to take this case into consideration later on).

**Comment by guiwitz** [comment]
A channel problem was preventing a correct segmentation. This is now fixed via PR #6.
Also the change implemented in https://github.com/guiwitz/napari-serialcellpose/commit/59101534401d5f92f7d5f2fa23a405a0b0bf9530 allows now to use thresholds with values from -10 to 10. From https://github.com/MouseLand/cellpose/issues/71, it is clear that the ```cellprob_threshold``` can take values roughly within this range. It is not fully clear what range of values are meaningful for ```flow_threshold```.

Note that in the napari-serialcellpose plugin both ```cellprob_threshold``` and ```flow_threshold``` that can be set in the interface correspond to the **same options** available when using cellpose programmatically. This is unlike the "model match threshold" option available in the cellpose-napari plugin which is related to ```flow_threshold``` via the formula visisble in https://github.com/MouseLand/cellpose-napari/blob/cb8424db78d49e36c2c41b704960a66a5b648689/cellpose_napari/_dock_widget.py#L64.
